src/python/py/models/README.md

```diff
@@ -0,0 +1,103 @@
+```
+# From wheel:
+python -m onnxruntime_genai.models.builder -m path_to_local_folder_on_disk -o /path/to/output/folder -p precision -e execution_provider
+
+# From source:
+python builder.py -m path_to_local_folder_on_disk -o /path/to/output/folder -p precision -e execution_provider
+```
+
+### Extra Options
+This scenario is for when you want to have control over some specific settings. The below example shows how you can pass key-value arguments to `--extra_options`.
+```
+# From wheel:
+python -m onnxruntime_genai.models.builder -m model_name -o /path/to/output/folder -p precision -e execution_provider -c cache_dir_to_save_hf_files --extra_options filename=decoder.onnx
+
+# From source:
+python builder.py -m model_name -o /path/to/output/folder -p precision -e execution_provider -c cache_dir_to_save_hf_files --extra_options filename=decoder.onnx
+```
+To see all available options through `--extra_options`, please use the `help` commands in the `Full Usage` section above.
+
+### Unit Testing Models
+This scenario is where your PyTorch model is already downloaded locally (either in the default Hugging Face cache directory or in a local folder on disk). If it is not already downloaded locally, here is an example of how you can download it.
+
+```
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+model_name = "your_model_name"
+cache_dir = "cache_dir_to_save_hf_files"
+
+model = AutoModelForCausalLM.from_pretrained(model_name, cache_dir=cache_dir)
+model.save_pretrained(cache_dir)
+
+tokenizer = AutoTokenizer.from_pretrained(model_name, cache_dir=cache_dir)
+tokenizer.save_pretrained(cache_dir)
+```
+
+#### Option 1: Use the model builder tool directly
+This option is the simplest but it will download another copy of the PyTorch model onto disk to accommodate the change in the number of hidden layers.
+```
+# From wheel:
+python -m onnxruntime_genai.models.builder -m model_name -o /path/to/output/folder -p precision -e execution_provider --extra_options num_hidden_layers=4
+
+# From source:
+python builder.py -m model_name -o /path/to/output/folder -p precision -e execution_provider --extra_options num_hidden_layers=4
+```
+
+#### Option 2: Edit the config.json file on disk and then run the model builder tool
+
+1. Navigate to where the PyTorch model and its associated files are saved on disk.
+2. Modify `num_hidden_layers` in `config.json` to your desired target (e.g. 4 layers).
+3. Run the below command for the model builder tool.
+
+```
+# From wheel:
+python -m onnxruntime_genai.models.builder -m model_name -o /path/to/output/folder -p precision -e execution_provider -c cache_dir_where_hf_files_are_saved
+
+# From source:
+python builder.py -m model_name -o /path/to/output/folder -p precision -e execution_provider -c cache_dir_where_hf_files_are_saved
+```
```

Comment:
Coud you please add a section to explain how user can modify the script and support their customized model?

Reply:
I'll add it in another PR